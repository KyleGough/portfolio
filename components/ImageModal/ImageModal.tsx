import Image from 'next/image';
import React, { useState } from 'react';

interface ImageModalProps {
  src: string;
  alt: string;
}

export const ImageModal: React.FC<ImageModalProps> = ({ src, alt }) => {
  const [open, setOpen] = useState(false);

  const openModal = () => setOpen(true);

  const closeModal = () => setOpen(false);

  return (
    <>
      <figure className="col-span-12 md:col-span-6 lg:col-span-4 relative">
        <div
          onClick={openModal}
          className="cursor-pointer border-2 border-link hover:border-link-hover h-40 overflow-hidden"
        >
          <Image
            src={src}
            alt={alt}
            objectFit="cover"
            objectPosition="top"
            width={300}
            height={150}
          />
        </div>
        <figcaption className="my-2 text-sm text-center text-light font-thin">
          {alt}
        </figcaption>
      </figure>
      {open && (
        <div
          role="dialog"
          onClick={closeModal}
          className="fixed bg-opacity-75 h-screen w-full left-0 top-0 flex justify-center items-center bg-black z-10"
        >
          <div className="bg-white rounded shadow-lg">
            <div className="p-1 items-center justify-center text-center">
              <img
                loading="lazy"
                className="text-center border-2 border-link max-w-full lg:max-w-screen-lg"
                src={src}
                alt={alt}
              />
            </div>
            <div className="items-center text-center w-100 border-t p-3">
              <p className="my-1 text-sm text-center font-thin">{alt}</p>
            </div>
          </div>
        </div>
      )}
    </>
  );
};
