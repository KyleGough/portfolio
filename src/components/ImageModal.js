import React, { useState } from 'react';

export default function ImageModal(props) {
    const [open, setOpen] = useState(false);

    const openModal = () => {
        setOpen(true);
    }

    const closeModal = () => {
        setOpen(false);
    }

    return (
        <>
        <div className='col-span-12 md:col-span-6 lg:col-span-4'>
            <img onClick={openModal} className='cursor-pointer border-2 border-link hover:border-link-hover' src={props.src} alt={props.caption} />
            <p className='my-2 text-sm text-center text-light font-thin'>{props.caption}</p>
        </div>
        {open ? 
            <div onClick={closeModal} className='fixed bg-opacity-75 h-screen w-full left-0 top-0 flex justify-center items-center bg-black'>
                <div className='bg-white rounded shadow-lg'> 
                    <div className='p-1 items-center justify-center text-center'>
                        <img className='text-center border-2 border-link max-w-full lg:max-w-screen-lg' src={props.src} alt={props.caption} />
                    </div>
                    <div className="items-center text-center w-100 border-t p-3">
                        <p className='my-1 text-sm text-center font-thin'>{props.caption}</p>
                    </div>
                </div>
            </div>
            : <></>
        }
        </>
    );
}