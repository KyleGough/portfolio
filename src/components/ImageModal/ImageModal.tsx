import React, { useState } from 'react';

interface ImageModalProps {
  src: string;
  caption: string;
}

export const ImageModal: React.FC<ImageModalProps> = ({ src, caption }) => {
  const [open, setOpen] = useState(false);

  const openModal = () => {
    setOpen(true);
  };

  const closeModal = () => {
    setOpen(false);
  };

  return (
    <>
      <figure className="col-span-12 md:col-span-6 lg:col-span-4">
        <img
          loading="lazy"
          onClick={openModal}
          className="cursor-pointer border-2 border-link hover:border-link-hover"
          src={src}
          alt={caption}
        />
        <figcaption className="my-2 text-sm text-center text-light font-thin">
          {caption}
        </figcaption>
      </figure>
      {open && (
        <div
          role="dialog"
          onClick={closeModal}
          className="fixed bg-opacity-75 h-screen w-full left-0 top-0 flex justify-center items-center bg-black"
        >
          <div className="bg-white rounded shadow-lg">
            <div className="p-1 items-center justify-center text-center">
              <img
                loading="lazy"
                className="text-center border-2 border-link max-w-full lg:max-w-screen-lg"
                src={src}
                alt={caption}
              />
            </div>
            <div className="items-center text-center w-100 border-t p-3">
              <p className="my-1 text-sm text-center font-thin">{caption}</p>
            </div>
          </div>
        </div>
      )}
    </>
  );
};
