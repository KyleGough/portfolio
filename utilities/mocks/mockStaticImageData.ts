const mockStaticImageData = {
  src: '/#image',
  width: 100,
  height: 100,
  blurDataURL: '/#blur',
};

export default mockStaticImageData;
