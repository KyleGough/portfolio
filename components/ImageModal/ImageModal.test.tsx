import mockStaticImageData from '@mocks/mockStaticImageData';
import { fireEvent, render, screen } from '@testing-library/react';
import React from 'react';

import { ImageModal } from './ImageModal';

describe('ImageModal component', () => {
  it('renders', () => {
    const { container } = render(
      <ImageModal image={mockStaticImageData} alt="Test Caption" />
    );
    expect(screen.getByText('Test Caption')).toBeVisible();
    expect(container).toMatchSnapshot();
  });

  it('dialog opens and closes on click', () => {
    render(<ImageModal image={mockStaticImageData} alt="Test Caption" />);
    expect(screen.getByText('Test Caption')).toBeVisible();

    // Modal should not be in the document by default.
    const img = screen.getByRole('img');
    expect(screen.queryByRole('dialog')).not.toBeInTheDocument();

    // Modal opens upon click event.
    fireEvent.click(img);
    expect(screen.queryByRole('dialog')).toBeInTheDocument();

    // Modal closes upon click event.
    const dialog = screen.getByRole('dialog');
    fireEvent.click(dialog);
    expect(screen.queryByRole('dialog')).not.toBeInTheDocument();
  });
});
