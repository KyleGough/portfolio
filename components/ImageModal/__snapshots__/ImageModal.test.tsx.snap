// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ImageModal component renders 1`] = `
<div>
  <figure
    class="col-span-12 md:col-span-6 lg:col-span-6"
  >
    <img
      alt="Test Caption"
      class="cursor-pointer border-2 border-link hover:border-link-hover"
      loading="lazy"
      src="/#"
    />
    <figcaption
      class="my-2 text-sm text-center text-light font-thin"
    >
      Test Caption
    </figcaption>
  </figure>
</div>
`;
